lifecycle-operator/apis/lifecycle/v1beta1/common/common.go

```diff
@@ -0,0 +1,216 @@
+package common
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"math/rand"
+	"strconv"
+
+	operatorcommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/common"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const WorkloadAnnotation = "keptn.sh/workload"
+const VersionAnnotation = "keptn.sh/version"
+const AppAnnotation = "keptn.sh/app"
+const PreDeploymentTaskAnnotation = "keptn.sh/pre-deployment-tasks"
+const PostDeploymentTaskAnnotation = "keptn.sh/post-deployment-tasks"
+const K8sRecommendedWorkloadAnnotations = "app.kubernetes.io/name"
+const K8sRecommendedVersionAnnotations = "app.kubernetes.io/version"
+const K8sRecommendedAppAnnotations = "app.kubernetes.io/part-of"
+const K8sRecommendedManagedByAnnotations = "app.kubernetes.io/managed-by"
+const PreDeploymentEvaluationAnnotation = "keptn.sh/pre-deployment-evaluations"
+const PostDeploymentEvaluationAnnotation = "keptn.sh/post-deployment-evaluations"
+const SchedulingGateRemoved = "keptn.sh/scheduling-gate-removed"
+const TaskNameAnnotation = "keptn.sh/task-name"
+const NamespaceEnabledAnnotation = "keptn.sh/lifecycle-toolkit"
+const CreateAppTaskSpanName = "create_%s_app_task"
+const CreateWorkloadTaskSpanName = "create_%s_deployment_task"
+const CreateAppEvalSpanName = "create_%s_app_evaluation"
+const CreateWorkloadEvalSpanName = "create_%s_deployment_evaluation"
+const AppTypeAnnotation = "keptn.sh/app-type"
+const KeptnGate = "keptn-prechecks-gate"
+const ContainerNameAnnotation = "keptn.sh/container"
+
+const MinKeptnNameLen = 80
+const MaxK8sObjectLength = 253
+
+var ErrCannotCastKeptnAppVersion = errors.New("cannot cast KeptnAppVersion to v1alpha3")
+var ErrCannotCastKeptnApp = errors.New("cannot cast KeptnApp to v1alpha3")
+
+type AppType string
+
+const (
+	AppTypeSingleService AppType = "single-service"
+	AppTypeMultiService  AppType = "multi-service"
+)
+
+// KeptnState  is a string containing current Phase state  (Progressing/Succeeded/Failed/Unknown/Pending/Cancelled)
+type KeptnState string
+
+const (
+	StateProgressing KeptnState = "Progressing"
+	StateSucceeded   KeptnState = "Succeeded"
+	StateFailed      KeptnState = "Failed"
+	StateUnknown     KeptnState = "Unknown"
+	StatePending     KeptnState = "Pending"
+	StateDeprecated  KeptnState = "Deprecated"
+	// StateCancelled represents state that was cancelled due to a previous step having failed.
+	// Deprecated: Use StateDeprecated instead. Should only be used in checks for backwards compatibility reasons
+	StateCancelled KeptnState = "Cancelled"
```

Comment:
i believe we can remove `StateCancelled` now, as we have deprecated that a couple of versions ago